resources/tests/test_resource_api.py

```diff
@@ -136,11 +137,41 @@ def test_non_public_resource_visibility(api_client, resource_in_unit, user):
     response = api_client.get(url)
     assert response.status_code == 200
     assert response.data['count'] == 1
-
     url = reverse('resource-detail', kwargs={'pk': resource_in_unit.pk})
     response = api_client.get(url)
     assert response.status_code == 200
 
+    # Authenticated as unit admin
+    user.unit_authorizations.authorized = staff_user
+    user.unit_authorizations.level = UnitAuthorizationLevel.admin
+    user.unit_authorizations.subject = resource_in_unit.unit
+    user.save()
```

Comment:
Hmm.. didn't notice that first, but this isn't adding a new authorization at all. Only reason it "works" is that the user is already a general admin. It doesn't verify anything though. So you should clear the `is_general_admin` flag and use the related manager to create the authorization:
```python
user.is_general_admin = False
user.save()
user.unit_authorizations.create(
    authorized=staff_user,
    level=UnitAuthorizationLevel.admin,
    subject=resource_in_unit.unit)
```